### PR TITLE
add HSV -> Color converter

### PR DESCRIPTION
**Pull request by sprsquish**
Add a static method to convert hue/saturation/brightness values to a 32-bit packed color value. This allows for extremely simple color cycling. Controlling brightness is also straightforward and can be applied to individual pixels rather than full strands.

The algorithm was adapted from here: http://www.cs.rit.edu/~ncs/color/t%5Fconvert.html

### CODE REVIEW

**Comment by zwass** [comment]
This seems like a useful feature...

@sprsquish have you abandoned it?

@ladyada @PaintYourDragon would you accept it if the above comments and merge conflicts were resolved?

I can adopt it if necessary.

**Comment by sprsquish** [comment]
@zwass Merge conflicts have been fixed. Any other comments?

File: Adafruit_NeoPixel.cpp [hunk 1]
```diff
@@ -1082,6 +1082,39 @@ uint32_t Adafruit_NeoPixel::Color(uint8_t r, uint8_t g, uint8_t b) {
   return ((uint32_t)r << 16) | ((uint32_t)g <<  8) | b;
 }
 
+// Convert Hue/Saturation/Brightness values to a packed 32-bit RBG color.
+// hue must be a float value between 0 and 360
+// saturation must be a float value between 0 and 1
+// brightness must be a float value between 0 and 1
+uint32_t Adafruit_NeoPixel::HSVColor(float h, float s, float v) {
```
Comment (zwass) [inline Adafruit_NeoPixel.cpp]: Maybe we could use the `constrain` function to ensure the preconditions suggested by this comment?

**Comment by zwass** [comment]
Mostly I'm just hoping one of the repo maintainers will take a look and merge your request ;)

File: Adafruit_NeoPixel.cpp (hunk 1 above)
Comment (sprsquish) [inline Adafruit_NeoPixel.cpp]: Oh, cool. Done.

**Comment by patmoore** [comment]
Is this going to be merged?

**Comment by sprsquish** [comment]
No idea. How do we get the attention of @ladyada or @PaintYourDragon or whomever is maintaining this project?

**Comment by PaintYourDragon** [comment]
Guess that'd be me. I've been stalling as I try to figure a polite way to say "not likely to merge any functions using floating-point arguments because they simply don't fit on Trinket/Gemma," not wanting to sound gift-horse-in-mouth-y toward code contributions, because I do appreciate folks' input, and the code doesn't (yet) have any "roadmap" comments to steer toward particular solutions.

I have a fixed-point HSV function that could be rolled in, but been stalling on that too as I'd like to maintain feature parity between the NeoPixel and DotStar libraries on both Arduino (C++) and Raspberry Pi (Python & C), and it's one of those time-consuming "need to sit down for like a whole day and validate all the changes on all the libraries at the same time" things, and I haven't had one of those whole days free.

**Comment by sprsquish** [comment]
@PaintYourDragon that was a very polite way to say it. Makes sense, too. 

I have some evenings free and would be happy to help. Do you have some time to write up your thoughts and point me to the relevant repos/code?

**Comment by zwass** [comment]
Yes, thank you for that response @PaintYourDragon. I'd vastly prefer to hear a true and legitimate reason why this can't be merged over radio silence :)

Would be awesome if you and @sprsquish could work this out, definitely would be a cool feature to have.

Thanks!

**Comment by patmoore** [comment]
Maybe we could not make the perfect the enemy of the good? Maybe judicious use of ```#ifdef``` ?

Also since someone is looking: are my pull requests cool?

**Comment by PaintYourDragon** [comment]
@patmoore Not sure yet, haven't had time to take a good look at them. From the description, the 'expose the pin' thing set of some alarm bells, but I'd need to actually look at the implementation, might be fine.

Today though, DotStar for Python is kicking my butt...

**Comment by patmoore** [comment]
@PaintYourDragon Appreciate if you can eyeball them quicker than this PR :-) Look at #77 that has all the changes rolled together.

**Comment by auzias** [comment]
So useful and still not merged since Dec 16, 2014 :/

**Comment by BsAtHome** [comment]
To solve the floating point issue, see pull #114.

I've implemented an all-integer version with near perfect mapping and highly optimized. That should make the floating point argument moot.

**Comment by relaxing** [comment]
This would be really useful.

**Comment by PaintYourDragon** [comment]
Making a sweep through some old PRs today. Grateful for the contribution, but a different PR somewhere along the way implemented HSV as an integer function (ColorHSV()) that should be kinder to AVR and similar devices. Thank you.

**Pull request closed without merging**